tools/floodgate/js/ui.js

```diff
@@ -0,0 +1,103 @@
+import {
+  createTag,
+  getAnchorHtml,
+  getLinkOrDisplayText,
+  getPathFromUrl,
+  showButtons,
+} from '../../loc/utils.js';
+import {
+  createColumn,
+  getFloodgateUrl,
+} from './utils.js';
+
+function getSharepointStatus(doc, isFloodgate) {
+  let sharepointStatus = 'Connect to Sharepoint';
+  let hasSourceFile = false;
+  let modificationInfo = 'N/A';
+  if (!isFloodgate && doc && doc.sp) {
+    if (doc.sp.status === 200) {
+      sharepointStatus = `${doc.filePath}`;
+      hasSourceFile = true;
+      modificationInfo = `By ${doc.sp?.lastModifiedBy?.user?.displayName} at ${doc.sp?.lastModifiedDateTime}`;
+    } else {
+      sharepointStatus = 'Source file not found!';
+    }
+  } else if (isFloodgate && doc && doc.fg && doc.fg.sp) {
+    if (doc.fg.sp.status === 200) {
+      sharepointStatus = `${doc.filePath}`;
+      hasSourceFile = true;
+      modificationInfo = `By ${doc.fg.sp?.lastModifiedBy?.user?.displayName} at ${doc.fg.sp?.lastModifiedDateTime}`;
+    } else {
+      sharepointStatus = 'Floodgated file not found!';
+    }
+  }
+  return { hasSourceFile, msg: sharepointStatus, modificationInfo };
+}
+
+function updateProjectInfo(project) {
+  document.getElementById('project-url').innerHTML = `<a href='${project.sp}' title='${project.excelPath}'>${project.name}</a>`;
+}
+
+function createTableWithHeaders() {
+  const $table = createTag('table');
+  const $tr = createTag('tr', { class: 'header' });
+  $tr.appendChild(createColumn('Source URL', 'header'));
+  $tr.appendChild(createColumn('Source File', 'header'));
+  $tr.appendChild(createColumn('Source File Info', 'header'));
+  $tr.appendChild(createColumn('Floodgated Content', 'header'));
+  $tr.appendChild(createColumn('Floodgated File Info', 'header'));
+  $table.appendChild($tr);
+  return $table;
```

Comment:
Great to see `createTag`. FYI in milo, we don't prefix our dom element variables with `$`. Not sure if there's a rule against it, but it's not a pattern we follow

Reply:
Loc and Floodgate share similar UI and they will be ported over to use Preact soon. All of this UI specific code will be removed at that point. Jingle is currently working on creating the layout. In any case, I have removed the usage of `$`.